ConsoleApp1/Program.cs

```diff
@@ -232,8 +146,199 @@ static void SearchRecords(string path)
+					}
+				}
+
+            } else
+			{
+				Console.WriteLine("invalid input");
+			}
+        }
+
+		static void SortRecords(string path)
+		{
+
+            Func<(string firstName, string lastName, string phone), int, string> SortBy = 
+				(a, b) => b == 1 ? a.firstName : b == 2 ? 
+					a.lastName : a.phone;
+			Func<string, string, int, bool> sorter = (a, b, o) => o == 1 ?
+				string.Compare(a, b, StringComparison.OrdinalIgnoreCase) < 0 ? false : true : 
+				string.Compare(a, b, StringComparison.OrdinalIgnoreCase) < 0 ? true : false;
+
+            (string firstName, string lastName, string phone)[] rec = ReadPhoneBookRecords(path);
+
+            Console.WriteLine("\t1. first name");
+            Console.WriteLine("\t2. last name");
+            Console.WriteLine("\t3. phone number");
+            Console.Write("Sort by: ");
+            var pressedKey = Console.ReadKey();
+			int sortBy = 1;
+			switch (pressedKey.Key)
+			{
+				case ConsoleKey.D1:
+					sortBy = 1;
+					break;
+				case ConsoleKey.D2:
+					sortBy = 2;
+					break;
+				case ConsoleKey.D3:
+					sortBy = 3;
+					break;
+				default:
+					sortBy = -1;
+					break;
+			}
+
+			Console.WriteLine();
+            Console.WriteLine("\t1. ASC");
+			Console.WriteLine("\t2. DSC");
+            Console.Write("Sort order: ");
+
+            pressedKey = Console.ReadKey();
+            int order = 1;
+            switch (pressedKey.Key)
+            {
+                case ConsoleKey.D1:
+                    order = 1;
+                    break;
+                case ConsoleKey.D2:
+                    order = 2;
+                    break;
+                default:
+                    order = -1;
+                    break;
+            }
+
+			if(sortBy != -1 && order != -1)
+			{
+				for (int i = 1; i < rec.Length; i++)
+				{
+					for (int j = i; j > 0 && sorter(SortBy(rec[j - 1],sortBy), SortBy(rec[j], sortBy), order); j--)
+					{
+                        (string firstName, string lastName, string phone) temp = rec[j];
+						rec[j] = rec[j - 1];
+						rec[j - 1] = temp;
+                    }
+				}
+
+                string[] strRes = new string[rec.Length + 1];
+                strRes[0] = "FirstName,LastName,PhoneNumber";
+
+                for (int i = 0; i < rec.Length; i++)
+                {
+                    strRes[i + 1] = $"{rec[i].firstName},{rec[i].lastName},{rec[i].phone}";
+                }
+                File.WriteAllLines(path, strRes);
+            }
+        }
+        static (string firstName, string lastName, string phone)[] SortRecords(
+			(string firstName, string lastName, string phone)[] rec, 
+			int sortBy = 2, 
+			int order = 1)
+        {
+
+            Func<(string firstName, string lastName, string phone), int, string> SortBy =
+                (a, b) => b == 1 ? a.firstName : b == 2 ?
+                    a.lastName : a.phone;
+            Func<string, string, int, bool> sorter = (a, b, o) => o == 1 ?
```

Comment:
You should simplify your code as possible 
Func<string, string, int, bool> sorter = (a, b, o) =>  string.Compare(a, b, StringComparison.OrdinalIgnoreCase) < 0 && o == 1;